src/App.tsx

```diff
@@ -2,24 +1,57 @@
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = 42;
+const startingPage = 5;
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(startingPage);
+
+  const items = getNumbers(1, total);
+  const totalPages = Math.ceil(items.length / perPage);
+
+  const indexOfLastItem = currentPage * perPage;
+  const indexOfFirstItem = indexOfLastItem - perPage;
+  const currentItems = items.slice(indexOfFirstItem, indexOfLastItem);
+
+  function handleSelectChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    setPerPage(Number(event.target.value));
+    setCurrentPage(1);
```

Comment:
The 'handleSelectChange' function is correctly updating the 'perPage' state and resetting the 'currentPage' to 1. However, consider checking if the new 'perPage' value is different from the current one before setting the state to avoid unnecessary re-renders.

Reply:
Ok